File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/dagger/DataComponent.java
package com.dbottillo.mtgsearchfree.dagger;

import com.dbottillo.mtgsearchfree.ActivityScope;
import com.dbottillo.mtgsearchfree.view.activities.CardLuckyActivity;
import com.dbottillo.mtgsearchfree.view.activities.CardsActivity;
import com.dbottillo.mtgsearchfree.view.activities.DeckActivity;
import com.dbottillo.mtgsearchfree.view.activities.MainActivity;
import com.dbottillo.mtgsearchfree.view.activities.SearchActivity;
import com.dbottillo.mtgsearchfree.view.fragments.AddToDeckFragment;
import com.dbottillo.mtgsearchfree.view.fragments.DecksFragment;
import com.dbottillo.mtgsearchfree.view.fragments.LifeCounterFragment;
import com.dbottillo.mtgsearchfree.view.fragments.MainFragment;
import com.dbottillo.mtgsearchfree.view.fragments.SavedFragment;

import dagger.Component;

@ActivityScope
@Component(modules = {PresentersModule.class}, dependencies = {AppComponent.class})
public interface DataComponent {
    void inject(MainActivity activity);

    void inject(SearchActivity searchActivity);

    void inject(CardsActivity cardsActivity);

    void inject(CardLuckyActivity luckyActivity);

    void inject(DeckActivity deckActivity);

    void inject(MainFragment mainFragment);

    void inject(SavedFragment savedFragment);

    void inject(LifeCounterFragment lifeCounterFragment);

    void inject(DecksFragment decksFragment);

    void inject(AddToDeckFragment addToDeckFragment);

}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/component/DataComponent.kt
package com.dbottillo.mtgsearchfree.component

import com.dbottillo.mtgsearchfree.ActivityScope
import com.dbottillo.mtgsearchfree.modules.PresentersModule
import com.dbottillo.mtgsearchfree.view.activities.CardLuckyActivity
import com.dbottillo.mtgsearchfree.view.activities.CardsActivity
import com.dbottillo.mtgsearchfree.view.activities.CommonCardsActivity
import com.dbottillo.mtgsearchfree.view.activities.MainActivity
import com.dbottillo.mtgsearchfree.view.fragments.MainFragment
import dagger.Component

@ActivityScope
@Component(modules = arrayOf(PresentersModule::class), dependencies = arrayOf(AppComponent::class))
interface DataComponent {
    fun inject(activity: MainActivity)

    fun inject(cardsActivity: CardsActivity)

    fun inject(luckyActivity: CardLuckyActivity)

    fun inject(mainFragment: MainFragment)
}